### PR TITLE
Stft

### PR DESCRIPTION
**Pull request by keunwoochoi**

### CODE REVIEW

**Comment by keunwoochoi** [comment]
(oh damn, it's mixed with #56....)
- i'm quite confident with the stft improvement. test was added too.
- a new problem is, in the test, `'reflect'`  padding with shorter-than-`fft_len` signal is expected to fail, while `'constant'` padding with shorter-than-`fft_len` signal is expected to succeed. i haven't figured out how to configure this in pytest.

**Comment by hagenw** [comment]
If the test success depends on a combination of different parameters, you can also combine them yourself, e.g. change
```
@pytest.mark.parametrize('fft_len', [512])
@pytest.mark.parametrize('hop_len', [256])
@pytest.mark.parametrize('waveform', [
    (torch.randn(1, 100000)),
    (torch.randn(1, 2, 100000)),
    pytest.param(torch.randn(1, 100), marks=xfail(raises=RuntimeError)),
])
@pytest.mark.parametrize('pad_mode', [
    # 'constant',
    'reflect',
])
```
to
```
@pytest.mark.parametrize('waveform,pad_mode,fft_len,'hop_len'', [
    (torch.randn(1, 100000), 'constant', 512, 256),
    (torch.randn(1, 2, 100000), 'constant', 512, 256),
    (torch.randn(1, 100), 'constant', 512, 256),
    (torch.randn(1, 100000), 'reflect', 512, 256),
    (torch.randn(1, 2, 100000), 'reflect', 512, 256),
    pytest.param(torch.randn(1, 100), 'reflect', 512, 256, marks=xfail(raises=RuntimeError)),
])
```
Now you have to specify every case yourself.

File: torchaudio_contrib/functional.py [hunk 1]
```diff
@@ -33,6 +76,15 @@ def stft(signal, fft_len, hop_len, window,
 
     # (!) Only 3D, 4D, 5D padding with non-constant
     # padding are supported for now.
+
+    if signal.dim() == 2:
+        # This is added because otherwise F.pad does not work.
```
Comment (ksanjeevan) [inline torchaudio_contrib/functional.py]: I thought we were going to move away from adding explicit padding and letting `torch.stft` handle it. We talked about it [here](https://github.com/keunwoochoi/torchaudio-contrib/pull/34#discussion_r271429945) and [here](https://github.com/keunwoochoi/torchaudio-contrib/pull/34#discussion_r272166166). Is there a strong incentive to keep it?

Comment (keunwoochoi) [reply torchaudio_contrib/functional.py]: Oh, yes this is more like a temporary-fix-meanwhile.

**Comment by keunwoochoi** [comment]
@hagenw Thanks. I think we can do it by creating two classes that simply inherits from the test, I'll push it later.

File: torchaudio_contrib/functional.py (hunk 1 above)
Comment (ksanjeevan) [reply torchaudio_contrib/functional.py]: Ok fair enough then :)

**Comment by keunwoochoi** [comment]
On testing - because we're gonna rely on `torch.stft` for the padding, i won't worry about pad test too much here.

**Pull request merged**